aldryn_newsblog/views.py

```diff
@@ -65,7 +65,12 @@ class PreviewModeMixin(EditModeMixin):
     """
     def get_queryset(self):
         qs = super(PreviewModeMixin, self).get_queryset()
-        if not self.edit_mode:
+        # check if user can see unpublished items. this will allow to switch
+        # to edit mode instead of 404 on article detail page. CMS handles the
+        # permissions.
+        user = self.request.user
+        user_can_edit = user.is_staff or user.is_superuser
+        if not (self.edit_mode or user_can_edit):
```

Comment:
So far this feature is not implemented yet. This may be proposed as a feature for future releases.